assets/src/hooks/useDetour.ts

```diff
@@ -55,6 +55,8 @@ export const useDetour = (routePatternId: RoutePatternId) => {
           }
         }
       )
+    } else {
```

Comment:
Should this also be gated by `shouldUpdate`? It feels like the odds of a race condition here are pretty low, but not 0?

Reply:
`shouldUpdate` is necessary because of the race condition in the promise. This `if` / `else` statement will only be evaluated when the `useEffect` hook re-renders, at which point `shouldUpdate` in the previous promise will have been set to `false` by the cleanup of the old version of the hook. So I think it's fine.

Reply:
Yeah I wasn't sure if the race condition could also just happen if the browser resolved `useEffect`s in a funny order or something. But maybe that's not how it works? I do imagine you'd have to try in order to make this cause a problem (if it's even possible)

Reply:
My understanding is that when a given `useEffect` hook re-renders, you at least know that it has already run the cleanup for the previous iteration.